### PR TITLE
Add (some) missing tests

### PR DESCRIPTION
**Pull request by florian-huber**
Adding some missing unit tests...
Idea was not to achieve 100% completeness but to (quickly) fill some gaps ;)

Thereby also found:
- minor bug in ``add_precursor_mz``: fixed
- there are still several ``filtering`` functions which do not have their own unit test --> no time to work on it now --> #162

### CODE REVIEW

File: tests/test_reduce_to_number_of_peaks.py [hunk 1]
```diff
@@ -89,3 +89,10 @@ def test_reduce_to_number_of_peaks_desired_5_check_sorting():
 
     assert spectrum.peaks.intensities.tolist() == [5., 4., 3., 100., 2.], "Expected different intensities."
     assert spectrum.peaks.mz.tolist() == [10., 30., 40., 50., 60.], "Expected different peaks to remain."
+
+
+def test_empty_spectrum():
+    spectrum_in = None
+    spectrum = reduce_to_number_of_peaks(spectrum_in)
+
+    assert spectrum is None, "Expected differnt handling of None spectrum."
```
Comment (sverhoeven) [inline tests/test_reduce_to_number_of_peaks.py]:
```suggestion
    assert spectrum is None, "Expected different handling of None spectrum."
```

File: tests/test_repair_inchi_inchikey_smiles.py [hunk 2]
```diff
@@ -132,3 +132,10 @@ def test_repair_inchi_inchikey_smiles_various_inchi_entered_as_smiles():
         assert spectrum.get("inchi") == "InChI=" + inchi.replace("InChI=", "").replace('"', "")
         assert spectrum.get("inchikey") == ""
         assert spectrum.get("smiles") == ""
+
+
+def test_empty_spectrum():
+    spectrum_in = None
+    spectrum = repair_inchi_inchikey_smiles(spectrum_in)
+
+    assert spectrum is None, "Expected differnt handling of None spectrum."
```
Comment (sverhoeven) [inline tests/test_repair_inchi_inchikey_smiles.py]:
```suggestion
    assert spectrum is None, "Expected different handling of None spectrum."
```

File: tests/test_require_minimum_number_of_peaks.py [hunk 3]
```diff
@@ -101,3 +101,10 @@ def test_require_minimum_number_of_peaks_required_5_or_10():
 
     assert spectrum is None, "Did not expect the spectrum to qualify because the number of peaks (4) is less " \
                              "than the required number (10)."
+
+
+def test_empty_spectrum():
+    spectrum_in = None
+    spectrum = require_minimum_number_of_peaks(spectrum_in)
+
+    assert spectrum is None, "Expected differnt handling of None spectrum."
```
Comment (sverhoeven) [inline tests/test_require_minimum_number_of_peaks.py]:
```suggestion
    assert spectrum is None, "Expected different handling of None spectrum."
```

**Review by sverhoeven — APPROVED** [review]
Looks good to me.

Just some typos in test assertions

**Comment by florian-huber** [comment]
Thanks for checking Stefan!

**Pull request merged**